### ISSUE TITLE
80-bit floating-point math on ppc64le

### ISSUE THREAD

**Issue by JeremyRand** [ptitSeb/box64#520]
From [this comment](https://github.com/ptitSeb/box64/issues/242#issuecomment-1052656727) by @ptitSeb :

> SSE/SSE2 instruction are converted t the underling SIMD instruction. Same for x87 that are converted to Float and Double operation (ignoring 80bits operation for most part).

[According to](https://github.com/FEX-Emu/FEX/issues/2409#issuecomment-1432585400) @awilfox :

> There is quad-float support in hardware with ISA 3.0 (Power9).

It might be interesting to support 80-bit x87 math on POWER9 via quad-precision floating-point math, for a speed-up when the default [BOX64_X87_NO80BITS=0](https://github.com/ptitSeb/box64/blob/main/docs/USAGE.md#box64_x87_no80bits-) is used.

My understanding is that quad float is only hardware-accelerated in POWER9 and higher (so POWER8LE users would need to disable it), and is only aliased to `long double` on a subset of distros (e.g. recent Fedora, grep [Talospace](https://talospace.com/) for articles that mention "long double").

It looks to me like the GCC options you want are `-mfloat128 -mfloat128-hardware`, and then use the `__float128` data type instead of `long double` so that it works as expected on older distros.

**Comment by JeremyRand** [comment]
CC: @classilla

**Comment by ptitSeb** [comment]
converting 80bits to 128bits long double is, for now, not supported. It can be done in theory. The main issue would be to conversion 80btis to/from 128bits that needs to be written. Once this is exist,  a new `HAVE_128BITS` can be added to the code to handle that.
Note that it is seldom used, and all internal computation will still be done in 64bits, so not sure it's worth the effort.

**Comment by ptitSeb** [comment]
> Some examples of games and software utilizing the 80-bit precision:
> 
> * [Frontier: First Encounters (1995) crashes on non-x86 platforms dosbox-staging/dosbox-staging#2900 (comment)](https://github.com/dosbox-staging/dosbox-staging/issues/2900#issuecomment-1740280812)
> * [Allow extended precision floating point calculations to be done in software dosbox-staging/dosbox-staging#3125](https://github.com/dosbox-staging/dosbox-staging/issues/3125)
> * [civil engineering calculations](https://github.com/dosbox-staging/dosbox-staging/pull/3149#discussion_r1405303291)
> 
> RISC-V Q/L and IBM z also support 128-bit precision in hardware.

Those are all DOS program. Point is, on windows & linux, there is very little use of actual x87 80bits precisions. And on DOS era, it was many use as a trick to move 64bits values (wich box64/box86 support) or use to blit memory with 80bits values at a time (rarely used, but I remember seeing that 1 time, I think it was with carmaggedon).
It's even more rare on 64bits program, as SSE2 is mandatory and used instead of x87 for most float computations.

**Comment by ptitSeb** [comment]
Yes, box86 target older windows games. Still there are very few games that needs 80bits precision.

**Comment by JeremyRand** [comment]
It's really rather rude to have your bot demand that I show up here annually to comment on the self-evident fact that this is still not implemented.

**Comment by LukeShortCloud** [comment]
I'm sorry for the automated response.

We are a small team. Managing hundreds of GitHub Issues has been very challenging for us. We are unable to go through and retest everything on a regular basis.

I'll mark this as long-term to prevent further automated messages.

**Comment by LukeShortCloud** [comment]
For further context, we have found the majority of old issues are simply solved in newer Box64 versions. In my free time, I have personally spent hundreds of hours managing our GitHub issues manually. It's not sustainable.

**Comment by JeremyRand** [comment]
No worries, I wish you luck sorting through the backlog.
